src/components/styles/OlStyleFill.vue

```diff
@@ -1,56 +1,143 @@
-const styledObj = inject<Ref<Style | null> | null>("styledObj", null);
+// Inject possible nullable Ref objects from the parent component
+const style = inject<Ref<Style | null>>("style", ref(null));
+const circle = inject<Ref<CircleStyle | null>>("circle", ref(null));
+const styledObj = inject<Ref<Style | null>>("styledObj", ref(null));
 
+// Use a custom composable to convert props into an object
 const properties = usePropsAsObjectProperties(props);
 
-if (style != null && circle == null) {
-  // in style object
-  let fill = new Fill(properties);
-  style?.value?.setFill(fill);
-
-  const applyFill = () => {
-    style?.value?.setFill(new Fill());
-    fill = new Fill(properties);
-    style?.value?.setFill(fill);
-  };
-  watch(properties, () => {
-    applyFill();
-  });
+// Function to create a gradient fill
+const createGradientFill = (
+  gradient?: Gradient,
+  width: number = 256,
+  height: number = 256,
+): Fill => {
+  const gradientCanvas = document.createElement("canvas");
+  const ctx = gradientCanvas.getContext("2d");
 
-  watch(style, () => {
-    applyFill();
-  });
+  if (!ctx) throw new Error("Unable to get canvas context");
 
-  onMounted(() => {
-    style?.value?.setFill(fill);
-  });
+  gradientCanvas.width = width;
+  gradientCanvas.height = height;
 
-  onUnmounted(() => {
-    style?.value?.setFill(new Fill());
-  });
-} else if (circle != null) {
-  // in circle
-  const applyFilltoCircle = (color?: string) => {
+  let grad: CanvasGradient;
+
+  // Create the corresponding CanvasGradient object based on the gradient type
+  if (gradient) {
+    switch (gradient.type) {
+      case "linear":
+        grad = ctx.createLinearGradient(
+          gradient.x0,
+          gradient.y0,
+          gradient.x1,
+          gradient.y1,
+        );
+        break;
+      case "radial":
+        grad = ctx.createRadialGradient(
+          gradient.x0,
+          gradient.y0,
+          gradient.r0,
+          gradient.x1,
+          gradient.y1,
+          gradient.r1,
+        );
+        break;
+      case "conic":
+        // Conic gradients are not directly supported by the Canvas API, use a linear gradient as a fallback
+        grad = ctx.createLinearGradient(0, 0, width, height);
+        break;
+      default:
+        throw new Error("Unsupported gradient type");
+    }
+
+    // Add color stops to the gradient
+    gradient.colorStops.forEach(([offset, gradientColor]) => {
+      grad.addColorStop(offset, gradientColor);
+    });
+
+    ctx.fillStyle = grad;
+  } else {
+    ctx.fillStyle = properties.color ?? "transparent";
+  }
+
+  ctx.fillRect(0, 0, width, height);
+
+  const dataURL = gradientCanvas.toDataURL();
+  return new Fill({ color: { src: dataURL } });
+};
+
+// Function to apply fill style to a Style object
+const applyFillToStyle = () => {
+  if (style.value) {
+    const fill = properties.gradient
+      ? createGradientFill(properties.gradient)
```

Comment:
The call here currently results in the behaviour that the gradient is re-calculated everytime some prop changes. For example when zooming in / out, you can se the gradient changes.
In my opinion this behaviour is not the best and I would suggest to calculate the garient once the component is created instead of re-creating it.